core/toolbox.js

```diff
@@ -116,7 +116,6 @@ Blockly.Toolbox.prototype.init = function() {
   // Clicking on toolbox closes popups.
   Blockly.bindEventWithChecks_(this.HtmlDiv, 'mousedown', this,
       function(e) {
-        Blockly.DropDownDiv.hide();
```

Comment:
To check if this is the right behaviour, open a dropdown on the workspace and then click somewhere on the toolbox.